File: src/types/campaign.ts
export type CampaignStatus = 'draft' | 'active' | 'paused' | 'completed' | 'cancelled';
export type ApplicationStatus = 'pending' | 'accepted' | 'rejected' | 'withdrawn';

export interface CampaignBudget {
    total: number;
    spent: number;
    currency: 'KES' | 'USD';
}

export interface CampaignRequirements {
    minFollowers: number;
    niches: string[];
    platforms: string[];
    contentType: 'post' | 'story' | 'reel' | 'video' | 'blog';
    deliverables: string[];
}

export interface Campaign {
    id: string;
    brandId: string;
    brandName: string;
    brandLogo: string;
    title: string;
    description: string;
    status: CampaignStatus;
    budget: CampaignBudget;
    requirements: CampaignRequirements;
    startDate: Date;
    endDate: Date;
    applicationDeadline: Date;
    maxInfluencers: number;
    applicationsCount: number;
    acceptedCount: number;
    createdAt: Date;
    updatedAt: Date;
    metrics?: CampaignMetrics;
}

export interface CampaignMetrics {
    impressions: number;
    reach: number;
    engagement: number;
    clicks: number;
    conversions: number;
    roi: number;
}

export interface CampaignApplication {
    id: string;
    campaignId: string;
    influencerId: string;
    influencerName: string;
    influencerAvatar: string;
    status: ApplicationStatus;
    proposedRate: number;
    proposal: string;
    portfolio: string[];
    submittedAt: Date;
    reviewedAt?: Date;
    reviewNotes?: string;
}

export interface CreateCampaignData {
    title: string;
    description: string;
    budget: number;
    currency: 'KES' | 'USD';
    requirements: CampaignRequirements;
    startDate: Date;
    endDate: Date;
    applicationDeadline: Date;
    maxInfluencers: number;
}

File: api/models/campaign.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime


class CampaignCreate(BaseModel):
    title: str
    description: str
    budget: float = 0
    status: str = "draft"
    niches: List[str] = []
    requirements: Optional[Dict[str, Any]] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None


class CampaignUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    budget: Optional[float] = None
    status: Optional[str] = None
    niches: Optional[List[str]] = None
    requirements: Optional[Dict[str, Any]] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None


class CampaignApplicationCreate(BaseModel):
    pitch: Optional[str] = None


class ApplicationDecision(BaseModel):
    status: str  # "accepted" or "rejected"
